feat: show proxy environment variables in npm config list

MaxBlack-dev opened the pull request:
Fixes #4170

## Description
This PR makes 
pm config list display proxy-related environment variables when they are set.

## Changes
- Modified lib/commands/config.js to display HTTP_PROXY, HTTPS_PROXY, NO_PROXY and their lowercase variants when set
- Added test case in 	est/lib/commands/config.js to verify the new behavior
- Environment variables appear in a new ; environment-related config section

## Motivation
As reported in #4170, 
pm config list was not showing environment variables like HTTP_PROXY, NOPROXY, etc., even though these variables affect how npm runs. This made it difficult for users to see what proxy settings their environment has and could lead to mistakes.

## Testing
Manual testing confirms:
- When proxy environment variables are set, they appear in the output
- When they are not set, the section doesn't appear
- Both uppercase and lowercase variants are shown when present

The new test case verifies that environment variables are displayed correctly.

wraithgar commented on lib/commands/config.js:
```diff
@@ -350,6 +350,22 @@ ${defData}
+      const envVars = []
+
+      // Show proxy-related environment variables if they're set
+      const proxyEnvVars = ['HTTP_PROXY', 'HTTPS_PROXY', 'NO_PROXY', 'http_proxy', 'https_proxy', 'no_proxy']
```
This info should not live in the command.  If there are variables that can affect config then `@npmcli/config` should own that list.  Even if it's a manual list for now.  Ideally that info would be coupled to the actual parsing of those but I don't think we have that kind of situation possible right now, since those are parsed explicitly in our fetching library and not parsed as config in npm.

MaxBlack-dev replied:
Good point! I've moved the list of proxy environment variables to @npmcli/config/lib/definitions/index.js as a new proxyEnv export. This keeps the environment variable list alongside other config-related constants like nerfDarts and shorthands.

The config command now imports and uses proxyEnv from @npmcli/config/lib/definitions instead of maintaining its own hardcoded list.

wraithgar commented on workspaces/config/lib/definitions/index.js:
```diff
@@ -71,10 +71,22 @@ const nerfDarts = [
+// Environment variables that can affect npm's behavior but are not npm configs
+// These are shown in `npm config list` to help users understand their environment
+const proxyEnv = [
+  'HTTP_PROXY',
```
Technically any casing will be picked up by npm here https://github.com/npm/agent/blob/abcc65439978ba2fa377f3e76a9853fb08e62f88/lib/proxy.js#L16

To prevent big surprises in the outside chance someone has set `process.env.Http_proxy` this should be handled in the config list in a similar way to how it's parsed in the agent (and other libs).

MaxBlack-dev replied:
Great catch! I've updated the code to handle proxy environment variables case-insensitively, matching how the agent library does it in [proxy.js](vscode-file://vscode-app/c:/Users/husiv/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html). The implementation now:

Changed the [proxyEnv](vscode-file://vscode-app/c:/Users/husiv/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) list to use lowercase keys (matching the agent's PROXY_ENV_KEYS)
Modified [npm config list](vscode-file://vscode-app/c:/Users/husiv/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) to check environment variables by converting keys to lowercase before comparison
Uses a Set to deduplicate so if someone has both [HTTP_PROXY](vscode-file://vscode-app/c:/Users/husiv/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) and http_proxy set, it only shows one (the first found)
This ensures variables like Http_Proxy, [HTTP_PROXY](vscode-file://vscode-app/c:/Users/husiv/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html), http_proxy, etc. are all detected, preventing surprises for users who might have set proxy variables in different

wraithgar commented:
This looks good.  Are you comfortable enough with git to pull out the changes in the config workspace to their on commit?  Our versioning is automatic based on conventional commit messages and the one for this PR doesn't fit with that change in the workspace package.  It'd want to be something specific like `feat: export environment config variable names`.

If not we can do it but then the commit authorship wouldn't be wholly yours, if that matters to you.

Commit:
feat: export environment config variable names

Commit:
feat: show proxy environment variables in npm config list

Adds proxy-related environment variables to npm config list output
to help users understand their proxy configuration. Variables are
checked case-insensitively to match the agent library behavior.

Closes #4170

MaxBlack-dev commented:
> This looks good. Are you comfortable enough with git to pull out the changes in the config workspace to their on commit? Our versioning is automatic based on conventional commit messages and the one for this PR doesn't fit with that change in the workspace package. It'd want to be something specific like `feat: export environment config variable names`.
> 
> If not we can do it but then the commit authorship wouldn't be wholly yours, if that matters to you.

I've split this into two commits:

1. eat: export environment config variable names - Exports the proxyEnv list from @npmcli/config
2. eat: show proxy environment variables in npm config list - Uses that list in the main package

This allows proper versioning for the config workspace package.

wraithgar commented:
Thanks!  The config list snapshot changes seem to be something else that fell through the cracks but they're not enough to block merging this PR.
